### ISSUE TITLE
Add a way to convert json to Message/Embeds

### ISSUE THREAD

**Issue by OcelusPRO** [discord-jda/JDA#2470]
### General Troubleshooting

- [X] I have checked for similar issues on the Issue-tracker.
- [X] I have updated to the [latest JDA version](https://ci.dv8tion.net/job/JDA/lastSuccessfulBuild/)
- [X] I have checked the branches or the maintainers' PRs for upcoming features.

### Feature Request

As indicated in the title, the idea would be to add a way to convert a json to some entity in JDA

There is already the EntityBuilder in JDA which allows to create some entities, however, this is not considered as an "official" way to do it and does not support the set of entities that might be useful

A simple use case would be to make it much easier to store custom embeds for servers to see complete messages (containing embeds and components)

Let's imagine for example a bot that allows to send personalized messages, a user who wants to create his personalized message could simply provide it in json (retrieve from a site such as discohook for example) to then be sent with the bot (which would check the validity of json etc)
### Example Use-Case

```java
public MessageEmbed createEmbed(String json, JDA jda){
   return jda.createEmbedFromJson(json)
}
```

We can imagine an equivalent for other types of entities such as messages, modals and other elements that could have a rendering to users

**Comment by Andre601** [comment]
There are internals iirc that cam be used. And this would be a very niche use case imo.

**Comment by DxsSucuk** [comment]
Sounds great!
As addition a way to export the Entity as json?
Otherwise we would need to use something like GSON for that and it could lead to inconsistency?

**Comment by Andre601** [comment]
I recall Minn mentioning a reason as to why conversion from/to json isn't exposed on a public level in the API, but I can't recall what it was.

**Comment by MinnDevelopment** [comment]
It is unlikely that we can support full message objects, simply because message features vary depending on the endpoint they utilize. For instance, `POST/channels/:channel_id/messages` supports `nonce` and `stickers`, while other endpoints do not.

However, supporting embeds seems to be reasonable. I've opened a PR for embed deserialization in #2471. This will allow you to create an instance of `EmbedBuilder` from a `DataObject`, which can parse both JSON and ETF.

A lot of other entities in JDA already support this ser/de through similar `toData()` and `fromData` methods.

**Comment by OcelusPRO** [comment]
> It is unlikely that we can support full message objects, simply because message features vary depending on the endpoint they utilize. For instance, POST/channels/:channel_id/messages supports nonce and stickers, while other endpoints do not.



Can't we imagine partial support for messages by removing non-general elements by default?

**Comment by DxsSucuk** [comment]
Or support messages fully and just ignore the unsupported parts internaly?

**Comment by MinnDevelopment** [comment]
Supporting messages is too inconsistent and confusing. It would also not be capable to support attachments due to them not being JSON representable.

**Comment by DxsSucuk** [comment]
Would it be able to store attachments as a base64 string?

**Comment by DxsSucuk** [comment]
Shouldn't this be closed since the PR got merged or will it stay open till next release?

**Comment by MinnDevelopment** [comment]
We usually close issues on releases.

**Comment by MinnDevelopment** [comment]
Resolved with [5.0.0-beta.11](https://github.com/discord-jda/JDA/releases/tag/v5.0.0-beta.11)
